core/models.py

```diff
@@ -581,11 +581,18 @@ def activate(self, drop_inactive_table=False):
             self.active = True
             self.save()
 
-    def deactivate(self, drop_table=False):
-        if drop_table:
-            self.delete_data_table()
-        self.active = False
-        self.save()
+    def deactivate(self, drop_table=False, activate_most_recent=False):
+        with transaction.atomic():
+            if activate_most_recent and self.active:
+                most_recent = self.table.data_tables.exclude(id=self.id).inactive().most_recent()
+                if most_recent:
+                    most_recent.activate(drop_inactive_table=drop_table)
```

Comment:
Se `drop_table` for `True`, o `activate` do `most_recent` vai tentar deletar a tabela inativa (`self`, nesse caso), mas tem também um `if drop_table` que tentará deletá-la (e daí vai dar erro), não?